### PR TITLE
[Engine] [CLIENT] Support Get Job Information from client

### PR DESCRIPTION
**Pull request by Hisoka-X**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

File: seatunnel-engine/seatunnel-engine-core/src/main/java/org/apache/seatunnel/engine/core/protocol/codec/SeaTunnelGetJobInfoCodec.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import static com.hazelcast.client.impl.protocol.codec.builtin.FixedSizeTypesCodec.encodeLong;
+
+import com.hazelcast.client.impl.protocol.ClientMessage;
+import com.hazelcast.client.impl.protocol.Generated;
+import com.hazelcast.client.impl.protocol.codec.builtin.DataCodec;
+
+/*
+ * This file is auto-generated by the Hazelcast Client Protocol Code Generator.
+ * To change this file, edit the templates or the protocol
+ * definitions on the https://github.com/hazelcast/hazelcast-client-protocol
```
Comment (EricJoy2048) [inline seatunnel-engine/seatunnel-engine-core/src/main/java/org/apache/seatunnel/engine/core/protocol/codec/SeaTunnelGetJobInfoCodec.java:36]: remove this comments.

File: seatunnel-engine/seatunnel-engine-client/src/test/java/org/apache/seatunnel/engine/client/SeaTunnelClientTest.java [hunk 2]
```diff
@@ -164,6 +169,37 @@ public void testGetJobMetrics() {
...
+
+        try {
+            final ClientJobProxy clientJobProxy = jobExecutionEnv.execute();
+            CompletableFuture.supplyAsync(clientJobProxy::waitForJobComplete);
+            long jobId = clientJobProxy.getJobId();
+
+            // Running
+            Assertions.assertNotNull(CLIENT.getJobInfo(jobId).getJobImmutableInformation());
+
+            await().atMost(30000, TimeUnit.MILLISECONDS)
```
Comment (EricJoy2048) [inline seatunnel-engine/seatunnel-engine-client/src/test/java/org/apache/seatunnel/engine/client/SeaTunnelClientTest.java]: 180000 is better.

File: seatunnel-engine/seatunnel-engine-core/src/main/java/org/apache/seatunnel/engine/core/protocol/codec/SeaTunnelGetJobInfoCodec.java (hunk 1 above)
Comment (Hisoka-X) [reply seatunnel-engine/seatunnel-engine-core/src/main/java/org/apache/seatunnel/engine/core/protocol/codec/SeaTunnelGetJobInfoCodec.java:36]: I checked other codec file, also have this comment, do I need remove it all?

Comment (EricJoy2048) [reply seatunnel-engine/seatunnel-engine-core/src/main/java/org/apache/seatunnel/engine/core/protocol/codec/SeaTunnelGetJobInfoCodec.java:36]: Another way is update this comment `* definitions on the https://github.com/hazelcast/hazelcast-client-protocol` to `seatunnel-engine/seatunnel-engine-core/src/main/resources/client-protocol-definition/SeaTunnelEngine.yaml`

**Review by EricJoy2048 — APPROVED** [review]

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/master/JobHistoryService.java [hunk 3]
```diff
@@ -176,6 +188,10 @@ private JobStateData toJobStateMapper(JobMaster jobMaster) {
         return new JobStateData(jobId, jobStatus, pipelineStateMapperMap);
     }
 
+    public void storeJobInfo(long jobId, JobDAGInfo jobInfo) {
+        finishedJobDAGInfoImap.put(jobId, jobInfo);
```
Comment (ic4y) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/master/JobHistoryService.java:192]: Suggest  set the data TTL time

Comment (Hisoka-X) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/master/JobHistoryService.java:192]: I will add it in the future.

**Review by ic4y — APPROVED** [review]
LGTM

**Pull request merged**